onnxscript/function_libs/torch_aten/ops/core.py

```diff
@@ -2726,15 +2726,73 @@ def aten_index_copy(
     raise NotImplementedError()
 
 
+@torch_op("aten::index_put")
 def aten_index_put(
-    self: TensorType,
-    indices: Optional[Sequence[TensorType]],
-    values: TensorType,
+    self: TReal,
+    indices: Sequence[TReal],
```

Comment:
Does indices need to be INT64?

Reply:
Looks like the doc says tensor(int32), tensor(int64)

Reply:
yes. you are right.